webapp/src/components/Metrics/PerformanceAnalysis.test.tsx

```diff
@@ -0,0 +1,138 @@
+  ];
+  const server = setupServer(...handlers);
+
+  beforeAll(() => server.listen());
+  afterEach(() => server.resetHandlers());
+  afterAll(() => server.close());
+
+  it("should have two toggle buttons for the train and test", () => {
+    renderPerformanceAnalysis({ train: true, eval: true });
+    // verify if the toggles displayed with correct names
+    expect(
+      screen.getByRole("button", { name: "Evaluation Set", pressed: true })
+    ).toHaveValue("eval");
+    expect(
+      screen.getByRole("button", { name: "Training Set", pressed: false })
+    ).toHaveValue("train");
+  });
+
+  it("should display the expected columns", async () => {
+    renderPerformanceAnalysis({ train: true, eval: true });
+    await waitFor(() => {
+      const expectedColumnHeaders = [
+        "Label",
+        "Total",
+        "Correct & Predicted",
+        "Correct & Rejected",
+        "Incorrect & Rejected",
+        "Incorrect & Predicted",
+        "Accuracy",
+        "Precision",
+        "Recall",
+        "F1",
+        "ECE",
+      ];
+      const actualColumnHeaders = screen.getAllByRole("columnheader");
+      expectedColumnHeaders.forEach((name, index) => {
+        name.includes("Correct")
+          ? expect(screen.getAllByTestId("CheckIcon")).toBeTruthy()
+          : name.includes("Incorrect")
+          ? expect(screen.getAllByTestId("XIcon")).toBeTruthy()
+          : expect(actualColumnHeaders[index].textContent).toEqual(name);
+      });
+    });
+  });
+
+  it("should display the default sort as desc for colum header 'Total number of utterances'", () => {
+    renderPerformanceAnalysis({ train: true, eval: true });
+    const header = screen.getByRole("columnheader", { name: /Total/ });
+    expect(header).toHaveAttribute("aria-sort", "descending");
+  });
+
+  it("should display the list of options if the user click on the column header 'filterValue'", () => {
+    renderPerformanceAnalysis({ train: true, eval: true });
+    fireEvent.mouseDown(screen.getByRole("button", { name: "Label" }));
+    const filterList = screen.getAllByRole("option");
+    expect(filterList).toHaveLength(10);
+
+    const expectedList = [
+      "Label",
+      "Prediction",
+      "Smart Tags",
+      "Extreme Length",
+      "Partial Syntax",
+      "Dissimilar",
+      "Almost Correct",
+      "Behavioral Testing",
+      "Pipeline Comparison",
+      "Uncertain",
+    ];
+    filterList.forEach((item, index) => {
+      expect(item.textContent).toEqual(expectedList[index]);
+    });
+
+    expect(screen.getByRole("option", { name: "Smart Tags" })).toHaveAttribute(
+      "aria-selected",
+      "false"
+    );
+    fireEvent.click(screen.getByRole("option", { name: "Prediction" }));
+    expect(
+      screen.getByRole("columnheader", { name: "Prediction" })
+    ).toBeInTheDocument();
+  });
+
+  it("should not display the Footer component if the number of rows are lesser than initial number", async () => {
```

Comment:
I think you mean
```suggestion
  it("should not display the Footer component if the number of rows is less than initial number", async () => {
```
but I'm not following. What initial number? The test doesn't manipulate anything, so I guess this is the initial number of rows, not less. Can you help me understand?

Reply:
I meant the same "INITIAL_NUMBER_VISIBLE = 5", as we mentioned in our logic. This test case will check if there should be no footer component displayed. The mock response that was added is at most four rows.

Reply:
Got it, thanks!